.github/workflows/ci.yml

```diff
@@ -44,3 +44,42 @@ jobs:
           pytest --cov=eventio --cov-report=xml
 
       - uses: codecov/codecov-action@v4
+
+  examples:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ["3.10", "3.11", "3.12"]
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v5
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Run examples
+        env:
+          PYTHON_VERSION: ${{ matrix.python-version }}
+        run: |
+          python --version
+          pip install -e '.[examples]'
+          cd examples
+          RESOURCES="../tests/resources/"
+
+          MPLBACKEND=agg
+          
+          # python iter_mc_events.py
```

Comment:
Why are these commented out? These should work on any simtel test file  in the resources

Reply:
I could not make them work unfortunately

some give StopIteration error, some EOF Error, etc

this one specifically for example gives me

```python
➜ python iter_mc_events.py ../tests/resources/gamma_100gev_1216.eventio
/Users/michele/Software/pyeventio/src/eventio/simtel/simtelfile.py:251: UnknownObjectWarning: object type encountered, which is no handled at the moment: RunHeader[1200]
  warnings.warn(
/Users/michele/Software/pyeventio/src/eventio/simtel/simtelfile.py:251: UnknownObjectWarning: object type encountered, which is no handled at the moment: TelescopeDefinition[1201]
  warnings.warn(
/Users/michele/Software/pyeventio/src/eventio/simtel/simtelfile.py:251: UnknownObjectWarning: object type encountered, which is no handled at the moment: EventHeader[1202](event_id=1)
  warnings.warn(
/Users/michele/Software/pyeventio/src/eventio/simtel/simtelfile.py:251: UnknownObjectWarning: object type encountered, which is no handled at the moment: ArrayOffsets[1203]
  warnings.warn(
/Users/michele/Software/pyeventio/src/eventio/simtel/simtelfile.py:251: UnknownObjectWarning: object type encountered, which is no handled at the moment: EventEnd[1209](event_id=1)
  warnings.warn(
/Users/michele/Software/pyeventio/src/eventio/simtel/simtelfile.py:251: UnknownObjectWarning: object type encountered, which is no handled at the moment: RunEnd[1210]
  warnings.warn(
Traceback (most recent call last):
  File "/Users/michele/Software/pyeventio/examples/iter_mc_events.py", line 17, in <module>
    with SimTelFile(args.inputfile) as f:
         ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/michele/Software/pyeventio/src/eventio/simtel/simtelfile.py", line 128, in __init__
    self.next_low_level()
  File "/Users/michele/Software/pyeventio/src/eventio/simtel/simtelfile.py", line 150, in next_low_level
    o = next(self)
        ^^^^^^^^^^
  File "/Users/michele/Software/pyeventio/src/eventio/base.py", line 115, in __next__
    read_sync_marker(self)
  File "/Users/michele/Software/pyeventio/src/eventio/base.py", line 183, in read_sync_marker
    check_size_or_raise(sync, constants.SYNC_MARKER_SIZE, zero_ok=True)
  File "/Users/michele/Software/pyeventio/src/eventio/base.py", line 166, in check_size_or_raise
    raise StopIteration
StopIteration
```

Reply:
ah with camsettings_v6.simtel.zst seems to work

Reply:
`plot_light_on_ground_by_particle.py` and `plot_light_on_ground_by_particle.py` tell me there is no `emitter` in files similar to the one I use for `plot_light_on_ground.py`

```python
➜ python examples/plot_light_on_ground_by_particle.py tests/resources/3_gammas_reuse_5.dat
Traceback (most recent call last):
  File "/Users/michele/Software/pyeventio/examples/plot_light_on_ground_by_particle.py", line 113, in <module>
    main()
  File "/Users/michele/Software/pyeventio/examples/plot_light_on_ground_by_particle.py", line 44, in main
    mask = np.isclose(emitter['mass'], mass)
                      ~~~~~~~^^^^^^^^
TypeError: 'NoneType' object is not subscriptable
```

`read_all_objects.py` instead returns this with many different files

```python
Traceback (most recent call last):
  File "/Users/michele/Software/pyeventio/examples/read_all_objects.py", line 16, in <module>
    o.parse()
  File "/Users/michele/Software/pyeventio/src/eventio/base.py", line 346, in parse
    raise ValueError('This object only contains subobjects')
ValueError: This object only contains subobjects
```

Reply:
> tell me there is no emitter in files 

We indeed have no file with `emitter` information. I'll produce one.

Reply:
I think the only one missing is read_all_objects.py which gives me the same error no matter the test file